xray/commands/audit/generic/auditmanager.go

```diff
@@ -154,88 +159,69 @@ func RunAudit(auditParams *Params) (results *Results, err error) {
 	return
 }
 
-func isEntitledForJas(serverDetails *config.ServerDetails) (entitled bool, xrayVersion string, err error) {
-	xrayManager, xrayVersion, err := commandsutils.CreateXrayServiceManagerAndGetVersion(serverDetails)
-	if err != nil {
-		return
-	}
-	if !version.NewVersion(xrayVersion).AtLeast(xrayutils.EntitlementsMinVersion) {
-		log.Debug("Entitlements check for ‘Advanced Security’ package failed:")
-		log.Debug(coreutils.MinimumVersionMsg, coreutils.Xray, xrayVersion, xrayutils.EntitlementsMinVersion)
+func isEntitledForJas(xrayManager *xray.XrayServicesManager, xrayVersion string) (entitled bool, err error) {
+	if e := coreutils.ValidateMinimumVersion(coreutils.Xray, xrayVersion, xrayutils.EntitlementsMinVersion); e != nil {
+		log.Debug(e)
 		return
 	}
 	entitled, err = xrayManager.IsEntitled(xrayutils.ApplicabilityFeatureId)
 	return
 }
 
-// genericAudit audits all the projects found in the given workingDirs
-func genericAudit(params *Params) *Results {
-	if err := coreutils.ValidateMinimumVersion(coreutils.Xray, params.xrayVersion, commandsutils.GraphScanMinXrayVersion); err != nil {
-		return NewAuditResults().SetAuditError(err)
-	}
-	log.Info("JFrog Xray version is:", params.xrayVersion)
-	log.Info("Scanning for vulnerable dependencies...")
-	if len(params.workingDirs) == 0 {
-		return doAudit(params)
-	}
-
-	return auditMultipleWorkingDirs(params)
-}
-
-func auditMultipleWorkingDirs(params *Params) *Results {
-	projectDir, err := os.Getwd()
-	if err != nil {
-		return NewAuditResults().SetAuditError(errorutils.CheckError(err))
+func runScaScan(params *Params) (results *Results) {
+	results = NewAuditResults()
+	rootDir, err := os.Getwd()
+	if errorutils.CheckError(err) != nil {
+		return results.SetAuditError(err)
 	}
-	defer func() {
-		err = errors.Join(err, os.Chdir(projectDir))
-	}()
-	results := NewAuditResults()
 	for _, wd := range params.workingDirs {
```

Comment:
In case we don't supply a specific `workingDirs`, does the root dir is a part of the `workingDirs` slice?